architecture/README.md

```diff
@@ -0,0 +1,190 @@
+        }
+    }
+
+    override fun onResume(scope: CoroutineScope) {
+        super.onResume(scope)
+
+        scope.launch {
+            // Do stuff that happens only when viewmodel is active
+        } 
+    }
+
+}
+```
+
+On Android the AndroidX `ViewModel` serves as a base. The Viewmodel lifecycle matches onResume/onPause of the `Activity` or `Fragment` that created it.
+To achieve this, the View should be bound using the `KalugaViewModelLifecycleObserver.bind()` method.
+For convenience default implementations for `Activity`, `Fragment`, and `DialogFragment` exist (`KalugaViewModelActivity`, `KalugaViewModelFragment`, and `KalugaViewModelDialogFragment` respectively).
+Alternatively the user can call `didResume()` and `didPause()` on the ViewModel manually in the `onResume()` and `onPause()` methods.
+
+On iOS automatic setup was omitted due to ObjC classes requiring to be final.
+To use it, the user must manually call the ViewModels `didResume`/`didPause` in `viewDidAppear`/`viewDidDisappear`.
+
+## Observables
+Kaluga supports `Observables` (one way binding) and `Subjects` (Two way binding). An Object can be created through a `ReadOnlyProperty` (making it immutable on both sides), a `Flow` (allowing the flow to modify the observer), or a `BaseFlowable` (allowing both the Flow and the owner of BaseFlowable to modify the observer.
+Subjects can be created using either a `ObservableProperty` or `BaseFlowable`. All these can easily be converted using `asObservable()` or `asSubject()` respectively.
+Observable values can be accessed through delegation. To account for the difference between empty values and optional values a `ObservableResult` containing either the value or a `Nothing` type is returned.
+
+```kotlin
+class SomeViewModel : BaseViewModel() {
+    private val flow = flowOf(1, 2, 3)
+    val flowObservable = flow.toObservable(coroutineScope)
+
+    val flowable: BaseFlowable<Int> = // someFlowable
+    val flowableSubject = flowable.toSubject(coroutineScope)
+
+    fun readValue(defaultValue: Int): Int? {
+        val currentFlowResult: ObservableResult<Int> by flowObservable
+        val currentFlowValue: Int? by  currentFlowResult
+        return currentFlowValue
+    }
+
+    fun postValue(value: Int) {
+        var currentSubjectResult: ObservableResult<Int> by flowableSubject
+        currentSubjectResult = ObservableResult.Result(value) 
+    }
+}
+```
+
+On Android both Observable and Subject are easily converted into `LiveData` objects (Subject being `MutableLiveData`), allowing lifecycle-aware binding.
+
+```kotlin
+val observable: Observable<Int>
+val liveData = observable.liveData
+
+val subject: Subject<Int>
+val mutableLiveData = subject.liveData
+
+init {
+    liveData.observe(lifeCycle, Observer { value ->
+        mutableLiveData.postValue(value)
+    })
+}
+```
+
+On iOS no such option exists.
+Instead observing works similar to `ReactiveX` solutions. Value changes can be observed using `observe(onNext: ...))`.
```

Comment:
nit: ReactiveX implementation on iOS is called RxSwift and most people are familiar with that name, not the ReactiveX

Reply:
Removed reference